subsys/net/ip/net_context.c

```diff
@@ -2837,8 +2876,13 @@ static enum net_verdict net_context_raw_packet_received(
 {
 	struct net_context *context = find_context(conn);
 
-	NET_ASSERT(context);
-	NET_ASSERT(net_pkt_iface(pkt));
+	if (context == NULL) {
```

Comment:
So while I'm ok with and support getting rid of asserts in public APIs (the function should fail, not crash the application), I'm not convinced we should do this for internal (static) functions. Here we have full control of the provided arguments, and providing incorrect ones most likely means we have a bug somewhere in our own code. IMO leaving asserts in such cases makes them easier to identify.

Reply:
Makes sense, I will rework the PR.